server/utils/scraping.js

```diff
@@ -0,0 +1,122 @@
+			pageData = await page.evaluate(() => {
+				const priceBeginning = /[a-zA-Z]*/;
+				const dollarSign = /[$]/;
+				//nothing inside or outside of page.evaluate is available outside of it's closure, because it is running purely in the dom
+				let titleData = document.getElementById(`productTitle`);
+				let priceData = document.getElementById(`price_inside_buybox`);
+				let imageData = document.getElementById(`landingImage`);
+				// let descriptionData = document.getElementById(`productDescription`);
+				// let descriptionData = [...document.querySelectorAll(`#feature-bullets ul li span`)].map(
+				// 	(elem) => elem.innerText
+				// );
+
+				let pageDataObject = {
+					title: titleData.innerText,
+					price: priceData.innerText.replace(priceBeginning, '').replace(dollarSign, ''),
+					image: imageData.src
+					// description: descriptionData.innerText
+				};
+				return JSON.stringify(pageDataObject);
+			});
+
+			await browser.close();
+			// return `page data: ${pageData}`;
+
+			console.log(pageData);
+		} else if (domainName(url) === 'ebay') {
+			//EBAY SECTION
+			await page.waitForSelector('#itemTitle');
+			pageData = await page.evaluate(() => {
+				const priceBeginning = /[a-zA-Z]*/;
+				const dollarSign = /[$]/;
+				let titleData = document.getElementById(`itemTitle`);
+				let titleDataText = titleData.innerText.split('').splice(16).join('');
+				let priceData = document.getElementById(`prcIsum`);
+				let imageData = document.getElementById(`viEnlargeImgLayer_img_ctr`);
+				let descriptionData = [...document.querySelectorAll(`#feature-bullets div div div`)].map(
+					(elem) => elem.innerText
+				);
+
+				let pageDataObject = {
+					title: titleDataText,
+					//will want to change this to just the number
+					price: priceData.innerText.replace(priceBeginning, ''),
+					image: imageData.src
+					// description: descriptionData
+				};
+				return JSON.stringify(pageDataObject);
+			});
+
+			await browser.close();
+			console.log(pageData);
+		} else if (domainName(url) === 'craigslist') {
+			//CRAIGSLIST SECTION
+			await page.waitForSelector('#titletextonly');
+			pageData = await page.evaluate(() => {
+				const priceBeginning = /[a-zA-Z]*/;
+				const dollarSign = /[$]/;
+				let titleData = document.getElementById(`titletextonly`);
+				let priceData = document.querySelector(`.price`);
+				let imageData = document.querySelector(`.swipe .swipe-wrap div img`);
+				// let descriptionData = [...document.querySelectorAll(`#feature-bullets div div div`)].map(
+				// 	(elem) => elem.innerText
+				// );
+				let descriptionData = document.querySelector('#postingbody');
+
+				let pageDataObject = {
+					title: titleData.innerText,
+					//will want to change this to just the number
+					price: priceData.innerText,
+					image: imageData.src,
+					description: descriptionData.innerText
+				};
+				return JSON.stringify(pageDataObject);
+			});
+
+			await browser.close();
+			// console.log(pageData);
```

Comment:
This I found in npm puppeteer docs - So you need to return a JS object?
```
const dimensions = await page.evaluate(() => {
    return {
      width: document.documentElement.clientWidth,
      height: document.documentElement.clientHeight,
      deviceScaleFactor: window.devicePixelRatio
    };
  });
 
  console.log('Dimensions:', dimensions);
 
  await browser.close();
```

Reply:
awesome I will try that!